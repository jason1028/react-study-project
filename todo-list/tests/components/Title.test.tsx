/**
 * @author jasonemchen
 * @desc 组件快照测试
*/
import React from "react";
import { render } from "@testing-library/react";
import Title from "@/components/Title";

describe("Title test", () => {
    it("渲染大字体", () => {
        // baseElement为渲染内容
        const { baseElement } = render(<Title type="large" title="big font"/>);
        // 渲染出来建立大字体快照
        expect(baseElement).toMatchSnapshot();
    });

    it("渲染小字体", () => {
        // baseElement为渲染内容
        const { baseElement } = render(<Title type="small" title="small font"/>);
        // 渲染出来建立小字体快照
        expect(baseElement).toMatchSnapshot();
    });
});

