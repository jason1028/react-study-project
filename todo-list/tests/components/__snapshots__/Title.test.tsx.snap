// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title test 渲染大字体 1`] = `
<body>
  <div>
    <p
      style="font-size: 2em; color: red;"
    >
       
      big font
       
    </p>
  </div>
</body>
`;

exports[`Title test 渲染小字体 1`] = `
<body>
  <div>
    <p
      style="font-size: 0.5em; color: green;"
    >
       
      small font
       
    </p>
  </div>
</body>
`;
